pulsar-common/src/main/java/org/apache/pulsar/common/protocol/schema/SchemaHash.java

```diff
@@ -55,15 +82,118 @@ public static SchemaHash of(SchemaData schemaData) {
+            result = new SchemaHash(
+                    hashFunction.hashBytes(schemaBytes == null ? new byte[0] : schemaBytes), schemaType);
+        }
+        return result;
     }
 
     public byte[] asBytes() {
         return hash.asBytes();
     }
+
+    private static class EmptySchemaHashFactory {
+        private static final HashCode EMPTY_HASH = hashFunction.hashBytes(new byte[0]);
+        private static final SchemaHash NONE_SCHEMA_HASH = new SchemaHash(EMPTY_HASH, NONE);
+        private static final SchemaHash STRING_SCHEMA_HASH = new SchemaHash(EMPTY_HASH, STRING);
+        private static final SchemaHash JSON_SCHEMA_HASH = new SchemaHash(EMPTY_HASH, JSON);
+        private static final SchemaHash PROTOBUF_SCHEMA_HASH = new SchemaHash(EMPTY_HASH, PROTOBUF);
+        private static final SchemaHash AVRO_SCHEMA_HASH = new SchemaHash(EMPTY_HASH, AVRO);
+        private static final SchemaHash BOOLEAN_SCHEMA_HASH = new SchemaHash(EMPTY_HASH, BOOLEAN);
+        private static final SchemaHash INT8_SCHEMA_HASH = new SchemaHash(EMPTY_HASH, INT8);
+        private static final SchemaHash INT16_SCHEMA_HASH = new SchemaHash(EMPTY_HASH, INT16);
+        private static final SchemaHash INT32_SCHEMA_HASH = new SchemaHash(EMPTY_HASH, INT32);
+        private static final SchemaHash INT64_SCHEMA_HASH = new SchemaHash(EMPTY_HASH, INT64);
+        private static final SchemaHash FLOAT_SCHEMA_HASH = new SchemaHash(EMPTY_HASH, FLOAT);
+        private static final SchemaHash DOUBLE_SCHEMA_HASH = new SchemaHash(EMPTY_HASH, DOUBLE);
+        private static final SchemaHash DATE_SCHEMA_HASH = new SchemaHash(EMPTY_HASH, DATE);
+        private static final SchemaHash TIME_SCHEMA_HASH = new SchemaHash(EMPTY_HASH, TIME);
+        private static final SchemaHash TIMESTAMP_SCHEMA_HASH = new SchemaHash(EMPTY_HASH, TIMESTAMP);
+        private static final SchemaHash KEY_VALUE_SCHEMA_HASH = new SchemaHash(EMPTY_HASH, KEY_VALUE);
+        private static final SchemaHash INSTANT_SCHEMA_HASH = new SchemaHash(EMPTY_HASH, INSTANT);
+        private static final SchemaHash LOCAL_DATE_SCHEMA_HASH = new SchemaHash(EMPTY_HASH, LOCAL_DATE);
+        private static final SchemaHash LOCAL_TIME_SCHEMA_HASH = new SchemaHash(EMPTY_HASH, LOCAL_TIME);
+        private static final SchemaHash LOCAL_DATE_TIME_SCHEMA_HASH = new SchemaHash(EMPTY_HASH, LOCAL_DATE_TIME);
+        private static final SchemaHash PROTOBUF_NATIVE_SCHEMA_HASH = new SchemaHash(EMPTY_HASH, PROTOBUF_NATIVE);
+        private static final SchemaHash BYTES_SCHEMA_HASH = new SchemaHash(EMPTY_HASH, BYTES);
+        private static final SchemaHash AUTO_SCHEMA_HASH = new SchemaHash(EMPTY_HASH, AUTO);
+        private static final SchemaHash AUTO_CONSUME_SCHEMA_HASH = new SchemaHash(EMPTY_HASH, AUTO_CONSUME);
+        private static final SchemaHash AUTO_PUBLISH_SCHEMA_HASH = new SchemaHash(EMPTY_HASH, AUTO_PUBLISH);
+
+        public static SchemaHash get(SchemaType schemaType) {
```

Comment:
How about using a ConcurrentHashMap and use computeIfAbsent to create new SchemaHash?

Reply:
I think there is overhead if using `ConcurrentHashMap` when multi-thread access.  Maybe we could have a test @Jason918

Reply:
Here is the test result, It seems that hard code is reasonable here,  which callded frequently
| | using ConcurrentHashMap| using swith case|
|--|---|---|
|1min |1491355.74 |1613411.62|
|2min|1517001.18|1613378.29|
|3min|1513898.52|1680711.18|
|4min|1490639.96 |1685437.97|